ffyaml/ffyaml_test.go

```diff
@@ -71,3 +71,75 @@ func TestParser(t *testing.T) {
 		})
 	}
 }
+
+func TestParserAllTypesEmptyVals(t *testing.T) {
+	t.Parallel()
+
+	for _, testcase := range []struct {
+		name string
+		file string
+		want fftest.Vars
+	}{
+		{
+			name: "basic empty vals",
+			file: "testdata/empty_basic_vals.yaml",
+			want: fftest.Vars{S: "", I: 0, F: 0.00, D: 0 * time.Second, B: false, X: nil},
+		},
+	} {
+		t.Run(testcase.name, func(t *testing.T) {
+			fs, testVars := fftest.EmptyValsFS()
+			testVars.ParseError = ff.Parse(fs, []string{},
+				ff.WithConfigFile(testcase.file),
+				ff.WithConfigFileParser(ffyaml.Parser),
+			)
+			pError := testVars.ParseError
+			checkParseErr(t, pError)
+
+			if err := fftest.Compare(&testcase.want, testVars); err != nil {
+				t.Fatal(err)
+			}
+		})
+	}
+}
+
+func checkParseErr(t *testing.T, pError error) {
+	if pError != nil {
+		var errMsg string
+		errMsg = "ParseError: " + pError.Error()
+		if errors.Is(pError, ff.StringConversionError{}) {
+			errMsg = errMsg + ";errType: " + "StringConversionError"
+		}
+		t.Fatal(errMsg)
+	}
+}
+
+func TestEmptyValsDontOverwritePresets(t *testing.T) {
+	t.Parallel()
+
+	for _, testcase := range []struct {
+		name string
+		file string
+		want fftest.Vars
+	}{
+		{
+			name: "preset FS vals not overwritten",
+			file: "testdata/empty_basic_vals.yaml",
+			want: fftest.Vars{S: "EMPTY_DEFAULT", I: -500000, F: 42.42,
+				D: 86400 * time.Second, B: true, X: []string{"strVal1", "strVal2"}},
```

Comment:
Please fix this.

Reply:
> Please fix this.

Fix what?  I am not clear on what you mean here.

Reply:
Sorry, that was ambiguous.

```suggestion
			want: fftest.Vars{
				S: "EMPTY_DEFAULT", 
				I: -500000, 
				F: 42.42,
				D: 86400 * time.Second, 
				B: true, 
				X: []string{"strVal1", "strVal2"},
			},
```

Reply:
OK, got it now - Sure, Formatting has been updated to enhance readability/clarity.

See Latest.